Rename of Ember.Router.router to Ember.Router.routerMicrolib

scalvert opened the pull request:
The is an intermediate clean-up PR en route to finishing the router service RFC (see what I did there 👀 ).

**Related PRs**: 
https://github.com/emberjs/website/pull/2806
https://github.com/emberjs/website/pull/2808

**Reviewers**
@rwjblue @locks @ef4 

**Changes**
- renamed `Ember.router` to `Ember.routerMicrolib` to disambiguate which router we're referring to. This will help avoid code like `this.router.router...`
- added deprecation warning for the property
- added tests for the deprecation warning

**How to test drive**
- Checkout this branch
- `ember serve` and ensure tests pass for both **Enable Opt Features** enabled and disabled

scalvert committed:
Renaming Ember.router to Ember._routerMicrolib to disambiguate
references.

rwjblue commented on packages/ember-routing/lib/system/router.js:
```diff
@@ -1563,4 +1564,9 @@ function representEmptyRoute(liveRoutes, defaultParentState, route) {
 
+deprecateProperty(EmberRouter.prototype, 'router', 'routerMicrolib', {
+  id: 'ember-router.router',
+  until: '2.16'
```
Can you add the `url` property here (from the entry you are adding to the deprecations page on the website)?

scalvert replied:
I will definitely add once it's done!

scalvert replied:
qq: how do i derive the URL from the section I added to the website? I see that there's a URL with a hash pointing to `http://emberjs.com/deprecations/v2.x/#toc_*`, but I'm not sure how it generates that URL. 

I've been trying to run bundler for the website, but keep getting errors installing deps...

rwjblue replied:
I merged the deprecation PR so that the site was auto-published:

http://emberjs.com/deprecations/v2.x/#toc_ember-router-router-renamed-to-ember-router-routermicrolib

scalvert replied:
👍

rwjblue commented on packages/ember-routing/tests/system/router_test.js:
```diff
@@ -255,3 +255,21 @@ QUnit.test('Router#triggerEvent ignores handlers that have not loaded yet', func
+  let router = createRouter();
+
+  expectDeprecation(function() {
+    equal(router.router, router.routerMicrolib);
```
Can you tweak to `assert.equal` here?

scalvert replied:
Yessir!

rwjblue commented:
LGTM, can you squash commits down?

krisselden commented:
Should it be prefixed with an underscore?

rwjblue commented:
@krisselden - Great point, and I can't believe I didn't think of that.  Yes, I think we should underscore it so that it is `router._routerMicrolib`...

@scalvert - Thoughts?

scalvert commented:
Yes to squashing. Yes to prefix. Good call @krisselden!

scalvert commented:
@rwjblue @krisselden renamed `routerMicrolib` to `_routerMicrolib` and squashed commits. Also updated the associated deprecations docs on the website (see second related PR in the description above)

rwjblue commented:
Awesome, thank you @scalvert!

stefanpenner commented:
This PR should likely have created a accessor that emits a deprecation for the `Router.router` property.

Some (debug related) workflows relied on `Router.router` to get a list of routes (for example: https://github.com/stefanpenner/ember-console/blob/master/lib/commands/console.js#L69)

Also https://github.com/emberjs/ember.js/issues/15235

Although this was private, ensuring we have a helpful error when accessing the property is most likely appropriate

rwjblue commented:
@stefanpenner - Please feel free to review the diff of this PR, it actually adds [tests for exactly that scenario](https://github.com/emberjs/ember.js/pull/14919/files#diff-2191e4723a68b9ef4a158268f721871eR259).
